spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/UnicastSendingMessageHandler.java

```diff
@@ -276,8 +297,10 @@ public void startAckThread() {
 	}
 
 	protected void send(DatagramPacket packet) throws Exception {
-		DatagramSocket socket = this.getSocket();
-		packet.setSocketAddress(this.getDestinationAddress());
+		DatagramSocket socket = getSocket();
+		if (this.socketExpression == null) {
+			packet.setSocketAddress(getDestinationAddress());
```

Comment:
This is another concern in pursuit of the previous mine in the `DatagramPacketMessageMapper`.
This one overrides the options populated from there:
```
public DatagramPacket(byte buf[], int offset, int length,
                          InetAddress address, int port) {
....
        setAddress(address);
        setPort(port);
}
```
When we use `headers`.
```
public synchronized void setSocketAddress(SocketAddress address) {
...
        setAddress(addr.getAddress());
        setPort(addr.getPort());
}
```
On this line.

Maybe `getDestinationAddress()` should deal with `requestMessage` headers as well, like we do in the Kafka Adapters? Then a new option really would be good under name `enableHeaderRouting` exactly here in the `UnicastSendingMessageHandler`.